packages/bitfield/src/lib/BitField.ts

```diff
@@ -0,0 +1,279 @@
+export class BitField<Flags extends Record<string, number> | Record<string, bigint>> {
+	public readonly type: Flags[keyof Flags] extends number ? 'number' : 'bigint';
+	public readonly zero: Flags[keyof Flags] extends number ? 0 : 0n;
+	public readonly mask: ValueType<this>;
+	public readonly flags: Flags;
+
+	public constructor(flags: Readonly<Flags>) {
+		if (typeof flags !== 'object' || flags === null) {
+			throw new TypeError('flags must be a non-null object');
+		}
+
+		const entries = Object.entries(flags) as [string, Flags[keyof Flags]][];
+		if (entries.length === 0) {
+			throw new TypeError('flags must be a non-empty object');
+		}
+
+		const type = typeof entries[0][1];
+		if (type !== 'number' && type !== 'bigint') {
+			throw new TypeError('type must be either "number" or "bigint"');
```

Comment:
Could have a better error message

Reply:
What do you suggest?

Reply:
`A bitfield can only use numbers or bigints for its values` or something... Also you really should do that check for every entry provided by the user

Reply:
It's done later, lines 25-48, does specific checks to ensure all values are the same type as the first one.

Reply:
Line 19 is not wrapped in a try-catch so this particular error can still be thrown. Please adjust the error message as requested by @vladfrangu

Reply:
I... already did.